plugins/lsp/language-servers/_mk-lsp.nix

```diff
@@ -179,6 +179,7 @@ in
       settings = lib.nixvim.mkSettingsOption {
         description = "The settings for this LSP.";
         options = settingsOptions;
+        submoduleType = true;
```

Comment:
Can we do this in the specific LSP servers, rather than all of them?

Reply:
> I would argue that it makes sense to have `submodule` `settings` options for LSP servers

I don't follow why this is the case for LSPs but not plugins.

Reply:
I don't have a strong opinion on this, but my reasoning was the following:
- It would be slightly annoying to set `submoduleType = true` for each server that needs it.
- I cannot think of an example of a language server that is configured with something else than an attrs.

Whereas for plugins, there are exceptions.

But I don't mind, if you prefer declaring on the server-basis, that's perfectly fine by me.

Reply:
I'm fine with it in this PR if the comment is framed as a TODO or FIXME. Especially as we plan to drop `plugins.lsp` entirely.